webapi/src/main/scala/org/knora/webapi/messages/admin/responder/projectsmessages/ProjectsMessagesADM.scala

```diff
@@ -505,130 +506,61 @@ case class ProjectADM(
+  }
 
-  /**
-   * Tries to return the value as an IRI.
-   */
-  def toIri: IRI =
-    maybeIri.getOrElse(
-      throw DataConversionException(s"Identifier $value is not of the required 'ProjectIdentifierType.IRI' type.")
-    )
+  case class Shortname(value: String) extends ProjectIdentifierADM {
+    def apply(value: String): ShortName = ShortName.make(value).fold(e => throw e.head, v => v)
+  }
 
-  /**
-   * Returns an optional value of the identifier.
-   */
-  def toIriOption: Option[IRI] =
-    maybeIri
+  case class Uuid(value: String) extends ProjectIdentifierADM
 
-  /**
-   * Tries to return the value as an SHORTNAME.
-   */
-  def toShortname: String =
-    maybeShortname.getOrElse(
-      throw DataConversionException(s"Identifier $value is not of the required 'ProjectIdentifierType.SHORTNAME' type.")
-    )
+  def asIriOption(id: ProjectIdentifierADM): Option[String] =
+    id match {
+      case Iri(value) => Some(value)
+      case _          => None
+    }
 
-  /**
-   * Returns an optional value of the identifier.
-   */
-  def toShortnameOption: Option[String] =
-    maybeShortname
+  def asShortcodeOption(id: ProjectIdentifierADM): Option[String] =
+    id match {
+      case Shortcode(value) => Some(value)
+      case _                => None
+    }
 
-  /**
-   * Tries to return the value as an SHORTCODE.
-   */
-  def toShortcode: String =
-    maybeShortcode.getOrElse(
-      throw DataConversionException(s"Identifier $value is not of the required 'ProjectIdentifierType.SHORTCODE' type.")
-    )
+  def asShortnameOption(id: ProjectIdentifierADM): Option[String] =
+    id match {
+      case Shortname(value) => Some(value)
+      case _                => None
+    }
 
-  /**
-   * Returns an optional value of the identifier.
-   */
-  def toShortcodeOption: Option[String] =
-    maybeShortcode
+  def asUuidOption(id: ProjectIdentifierADM): Option[String] =
+    id match {
+      case Uuid(value) => Some(s"http://rdfh.ch/projects/${value}")
+      case _           => None
+    }
 
   /**
-   * Returns the string representation
+   * Helper method that gets right identifier value.
+   *
+   * @param identifier either IRI, Shortname, Shortcode or UUID of the project.
+   * @return identifier's value as [[String]]
    */
-  override def toString: IRI =
-    s"ProjectIdentifierADM(${this.value})"
-
-}
-
-/**
- * Project identifier types:
- *  - IRI
- *  - Shortcode
- *  - Shortname
- */
-sealed trait ProjectIdentifierType
-object ProjectIdentifierType {
-  case object IRI       extends ProjectIdentifierType
-  case object SHORTCODE extends ProjectIdentifierType
-  case object SHORTNAME extends ProjectIdentifierType
+  def getId(identifier: ProjectIdentifierADM): String =
+    identifier match {
+      case Iri(value)       => value
+      case Shortname(value) => value
+      case Shortcode(value) => value
+      case Uuid(value)      => s"http://rdfh.ch/projects/${value}"
+    }
```

Comment:
>    * Helper method that gets right identifier value.

What is a "right identifier value"? 
Is it the IRI representation of the given identifier?

Reply:
Right in the meaning the one was passed as param, because we currently support 3 + the new one UUID which is a part of IRI...